### ISSUE TITLE
Appimage based games fail to launch if steam is run in a sandboxed mode.

### ISSUE THREAD

**Issue by Redhawk18** [ValveSoftware/steam-for-linux#11364]
#### Your system information

* Steam client version (build number or date): 1.0.0.74
* Distribution (e.g. Ubuntu): Nixos
* Opted into Steam client beta?: [Yes/No] No
* Have you checked for system updates?: [Yes/No] Yes
* Steam Logs: [generate by running this command in a terminal `tar -zcvf ~/Desktop/steam-logs.tar.gz ~/.steam/steam/logs`]
* GPU: AMD

#### Please describe your issue in as much detail as possible:
Any Love2d game to built as an appimage is not guaranteed to work on generic linux computers. Which is a cause for of refunding games.

#### Steps for reproducing this issue:
example game is [Gravity Circuit](https://store.steampowered.com/app/858710/Gravity_Circuit/)
1. launch button
2. Fails to launch on distros where appimage is not setup by default. (not steamos)

**Comment by Redhawk18** [comment]
Even when adding fuse, the app image still crashes
```
chdir "/mnt/games/SteamLibrary/steamapps/common/Gravity Circuit"
ERROR: ld.so: object '/home/redhawk/.local/share/Steam/ubuntu12_32/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS32): ignored.
ERROR: ld.so: object '/home/redhawk/.local/share/Steam/ubuntu12_64/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS64): ignored.
ERROR: ld.so: object '/home/redhawk/.local/share/Steam/ubuntu12_32/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS32): ignored.
ERROR: ld.so: object '/home/redhawk/.local/share/Steam/ubuntu12_32/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS32): ignored.
Adding process 9755 for gameID 858710
fusermount: mount failed: Operation not permitted
pid 9757 != 9756, skipping destruction (fork without exec?)

Cannot mount AppImage, please check your FUSE setup.
You might still be able to extract the contents of this AppImage 
if you run it with the --appimage-extract option. 
See https://github.com/AppImage/AppImageKit/wiki/FUSE 
for more information
open dir error: No such file or directory
Adding process 9756 for gameID 858710
Uploaded AppInterfaceStats to Steam
Removing process 9756 for gameID 858710
Removing process 9755 for gameID 858710
```

I would also like to add that if I browse to the folder and add the correct `steam_appid.txt` and run the AppRun, the game just works perfectly. The issue is that steam is sandboxed and fails to open the appimage. While I could use something like proton, the controller input through this engine with proton will not register controller input after clicking out of the window.

**Comment by Samueru-sama** [comment]
Doesn't NixOS use bubblewrap to run applications? in that case that's the problem. 

You might still be able to run appimages by setting the env variable `APPIMAGE_EXTRACT_AND_RUN=1` though

**Comment by Redhawk18** [comment]
> Doesn't NixOS use bubblewrap to run applications? in that case that's the problem.
> 
> You might still be able to run appimages by setting the env variable `APPIMAGE_EXTRACT_AND_RUN=1` though

Yes, and that doesn't work. It works perfectly without steam though. The problem is steam doesn't have the tools to deal with this. I already added libfuse and it still fails to mount.

**Comment by Samueru-sama** [comment]
> > Doesn't NixOS use bubblewrap to run applications? in that case that's the problem.
> > You might still be able to run appimages by setting the env variable `APPIMAGE_EXTRACT_AND_RUN=1` though
> 
> Yes, and that doesn't work. It works perfectly without steam though. The problem is steam doesn't have the tools to deal with this. I already added libfuse and it still fails to mount.

We use bubblewrap in this [Steam AppImage](https://github.com/ivan-hc/Steam-appimage) and we had to [patch it](https://github.com/Samueru-sama/Conty/blob/master/caps.patch) in order to let Steam launch AppImages from Steam.

You might wanna test building bubblewrap with that patch and also pass the `CAP_SYS_ADMIN` capability to Steam.

**Comment by Redhawk18** [comment]
> CAP_SYS_ADMIN

Doesn't this basically let it run as root? I remember it from when I had to setup sunshine

**Comment by Samueru-sama** [comment]
> > CAP_SYS_ADMIN
> 
> Doesn't this basically let it run as root? I remember it from when I had to setup sunshine

I'm really not sure, as far as I know Steam itself won't be running as root, also I think the security implications really depend if bubblewrap is suid or not. 

In the case of the appimage everything happens at user level and root is never needed for anything, I don't know about Nix.

**Comment by Redhawk18** [comment]
> > > CAP_SYS_ADMIN
> > 
> > 
> > Doesn't this basically let it run as root? I remember it from when I had to setup sunshine
> 
> I'm really not sure, as far as I know Steam itself won't be running as root, also I think the security implications really depend if bubblewrap is the suid version or not.
> 
> In the case of the appimage everything happens at user level and root is never needed for anything, I don't know about Nix.

I asked the people in my sphere who control that, but the dev created a extracted option in steam that just extracts the appimage which works perfectly.
